maas-api.yaml

```diff
@@ -564,28 +686,19 @@ paths:
         Rather than constructing this URL directly, you should get it
         from the `temporaryURL` property from a computed trip.
       parameters:
-        - name: v
-          in: query
-          type: integer
-          description: Requested version number of output. Needs to be 11.
-          minimum: 11
-          maximum: 11
-          multipleOf: 1
+        - name: id
+          in: path
+          schema:
+            type: string
           required: true
-        - name: DOCUMENTATION_NOT_COMPLETE
-          in: query
-          type: boolean
+          description: Identifier of trip as returned by previous server call.
       responses:
         200:
-          description: Successful response, same format as routing.json, but with only one trip.
-          schema:
-            $ref: '#/definitions/RoutingResponse'
-    parameters:
-      - name: id
-        in: path
-        type: string
-        required: true
-        description: Identifier of trip as returned by previous server call.
+          description: Successful response
+          content:
+            application/json:
+              schema:
+                $ref: '#/components/schemas/Trip'
 
   /trip/{id}/save:
```

Comment:
shall we remove `/save`, `/update`, and `/planned` from the API specs as they are in the response of `/trip/{id}`?  can we unify them to all have the same response schema?  for `/update`, instead of the `If-None-Match` header, we may need to include a query parameter in the `updateURL` response field?

Reply:
I'm not sure, I follow. They serve different use cases, so I do not see the advantage of unifying them.

Reply:
I added the `If-None-Match` as a replacement to passing a query parameter to use a more HTTP standardway of achieving the same thing.

Reply:
they do serve different use cases, but they all are actions on the same trip.  what i was saying before is that they are already specified in the response of other URLs so i wasn't sure if they need to be documented separately.  They do need to be documented separately if they have different response schemas, so I wanted to unify them.  Also they need to be documented if they require a special header or post body.

this is what i'm thinking now:

1. could we remove the saveURL, updateURL, plannedURL from the other responses and rely on developers to use the defined url format in the API?
2. can we change trip saving to be a PUT to `/persistentURLs` with a PUT body of `tripID`?  (sorry i don't know the URL scheme for persistent URLs)
3. can we merge trip updating with getting?  `If-None-Match` would then be documented under `/trip/{id}`
4. can we make trip planned to be a POST to `/trip/{id}`?  then we can have more actions in the future without creating a new URL for each action.

btw, if i save a trip, does the `updateURL` returned work forever?  (it's not clear from the docs)

Reply:
Ok, makes more sense now.

1. I don't see any issue with documenting these endpoints, even if trips already have the URL for them in their reply. It's good practice to document endpoints even if you don't construct the URLs themselves yourselves (which is the only info contained in the trips). There's a recommendation for APIs to provide follow-up URLs in the responses rather than developers creating URLs themselves.
2. Interesting idea though it means we can't just have the URL in the trip response anymore and let users GET on that. I guess you don't want both `GET trips/{id}/save` and `PUT /persistentURLs`?
3. Good idea, though we'd lose the information of whether the trip has any real-time data or not (via the presence of `updateURL`), but I don't mind that this then implicitly means that every trip could have real-time data.
4. I prefer not as I would read `POST /trip/{id}` as creating/storing a trip, not adding something to an existing trip, so I prefer the current format.

In general, I prefer many simple endpoints over fewer but more complex or overloaded endpoints. But that's also a discussion to be had with the MaaS Alliance working group.

Reply:
I've addressed the third point by merging `trip/{id}/update` into `trip/{id}`

Reply:
1. You are right, we should document all endpoints, even ones that are part of a hypermedia API.
2. You understood me correctly.  Just trying to get closer to CRUD, especially in the case where we are saving some data.
3. Thanks
4. How about `PATCH`?

Is there someone in MaaS Alliance we can/should tag in this pull request, so they have input into the proposed API?  Or is that too many chefs in the kitchen?

Reply:
I've merged in this PR here and submitted it as a new PR to the repo of the MaaS Alliance: https://github.com/maas-alliance/apis/pull/6. So all follow-up discussions or suggestions is best to go there, or if it's a specific proposal as a new PR in this repo.

On 2: I agree that `GET /trips/{id}/save` is not ideal and doesn't match CRUD well. On 4: I prefer to stick with `GET, POST, PUT, DELETE` as it gets a bit fuzzy when you use the HTTP verbs.

Anyway, there's also a call with the working group on Thursday. Will keep you posted if there's any news.